tests/operator_test.go

```diff
@@ -1979,18 +1979,20 @@ spec:
 				sanityCheckDeploymentsExist()
 
 				By("setting the right uninstall strategy")
-				kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				kv.Spec.UninstallStrategy = v1.KubeVirtUninstallStrategyBlockUninstallIfWorkloadsExist
-				_, err = virtClient.KubeVirt(kv.Namespace).Update(kv)
-				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() error {
+					kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+					kv.Spec.UninstallStrategy = v1.KubeVirtUninstallStrategyBlockUninstallIfWorkloadsExist
+					_, err = virtClient.KubeVirt(kv.Namespace).Update(kv)
```

Comment:
You don't have to re-execute the Get() as well, it's just the Update that can cause a conflict.

My recommendation would be to use `retry.RetryOnConflict` and only retry the Update().

```go
kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
					Expect(err).ToNot(HaveOccurred())

err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
    _, err = virtClient.KubeVirt(kv.Namespace).Update(kv)
    return err
})
Expect(err).ToNot(HaveOccurred())
```

https://pkg.go.dev/k8s.io/client-go/util/retry#RetryOnConflict

Reply:
If I try to `Update` with the same resource I will get the conflict every time, for this I think it is mandatory to fetch the updated one.
Also in the reference you shared the resource is fetched every time.
```
// Fetch the resource here; you need to refetch it on every try, since
// if you got a conflict on the last update attempt then you need to get
// the current version before making your own changes.
```
Also `RetryOnConflict` does not give the opportunity to specify a timeout.
WDYT?

Reply:
Ah, you're right, it was a long time since I used it, sorry just disregard the noise :)